proxysql/datadog_checks/proxysql/proxysql.py

```diff
@@ -0,0 +1,117 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from contextlib import closing, contextmanager
+
+import pymysql
+import pymysql.cursors
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+from datadog_checks.base.utils.db import QueryManager
+from datadog_checks.proxysql.queries import (
+    STATS_COMMAND_COUNTERS,
+    STATS_MEMORY_METRICS,
+    STATS_MYSQL_CONNECTION_POOL,
+    STATS_MYSQL_GLOBAL,
+    STATS_MYSQL_QUERY_RULES,
+    STATS_MYSQL_USERS,
+    VERSION_METADATA,
+)
+from datadog_checks.proxysql.ssl_utils import make_insecure_ssl_client_context, make_secure_ssl_client_context
+
+ADDITIONAL_METRICS_MAPPING = {
+    'command_counters_metrics': STATS_COMMAND_COUNTERS,
+    'connection_pool_metrics': STATS_MYSQL_CONNECTION_POOL,
+    'users_metrics': STATS_MYSQL_USERS,
+    'memory_metrics': STATS_MEMORY_METRICS,
+    'query_rules_metrics': STATS_MYSQL_QUERY_RULES,
+}
+
+
+class ProxysqlCheck(AgentCheck):
+
+    SERVICE_CHECK_NAME = "can_connect"
+    __NAMESPACE__ = "proxysql"
+
+    def __init__(self, name, init_config, instances):
+        super(ProxysqlCheck, self).__init__(name, init_config, instances)
+        self.host = self.instance.get("host", "")
+        self.port = int(self.instance.get("port", 0))
+        self.user = self.instance.get("username", "")
+        self.password = str(self.instance.get("password", ""))
+
+        if not all((self.host, self.port, self.user, self.password)):
+            raise ConfigurationError("ProxySQL host, port, username and password are needed")
+
+        self.tls_verify = self.instance.get("tls_verify", False)
+        self.validate_hostname = self.instance.get("validate_hostname", True)
+        self.tls_ca_cert = self.instance.get("tls_ca_cert", None)
+        self.base_tags = self.instance.get("tags", [])
+        self.connect_timeout = self.instance.get("connect_timeout", 10)
+        self.read_timeout = self.instance.get("read_timeout", None)
```

Comment:
Same here.

Reply:
Do I miss something? this seems correctly documented.

Reply:
Sorry github mishaved. Same as in "Remove None".

Reply:
Oh comments were not in order, got it 👍